core/services/fluxmonitorv2/flux_monitor_test.go

```diff
@@ -0,0 +1,1674 @@
+		DotDagSource: `
+// data source 1
+ds1 [type=http method=GET url="https://pricesource1.com" requestData="{\\"coin\\": \\"ETH\\", \\"market\\": \\"USD\\"}"];
+ds1_parse [type=jsonparse path="latest"];
+
+// data source 2
+ds2 [type=http method=GET url="https://pricesource1.com" requestData="{\\"coin\\": \\"ETH\\", \\"market\\": \\"USD\\"}"];
+ds2_parse [type=jsonparse path="latest"];
+
+ds1 -> ds1_parse -> answer1;
+ds2 -> ds2_parse -> answer1;
+
+answer1 [type=median index=0];					
+`,
+	}
+}
+
+func NewPipelineRun() fluxmonitorv2.PipelineRun {
+	jobID := int32(1)
+	pipelineRunner := new(pipelinemocks.Runner)
+	pipelineSpec := pipeline.Spec{
+		ID: 1,
+		DotDagSource: `
+// data source 1
+ds1 [type=http method=GET url="https://pricesource1.com" requestData="{\\"coin\\": \\"ETH\\", \\"market\\": \\"USD\\"}"];
+ds1_parse [type=jsonparse path="latest"];
+
+// data source 2
+ds2 [type=http method=GET url="https://pricesource1.com" requestData="{\\"coin\\": \\"ETH\\", \\"market\\": \\"USD\\"}"];
+ds2_parse [type=jsonparse path="latest"];
+
+ds1 -> ds1_parse -> answer1;
+ds2 -> ds2_parse -> answer1;
+
+answer1 [type=median index=0];					
+		`,
+	}
+	l := *logger.Default
+
+	return fluxmonitorv2.NewPipelineRun(
+		pipelineRunner,
+		pipelineSpec,
+		jobID,
+		l,
+	)
+}
+
+func TestFluxMonitor_PollIfEligible(t *testing.T) {
+	testCases := []struct {
+		name              string
+		eligible          bool
+		connected         bool
+		funded            bool
+		answersDeviate    bool
+		hasPreviousRun    bool
+		previousRunStatus pipeline.RunStatus
+		expectedToPoll    bool
+		expectedToSubmit  bool
+	}{
+		{
+			name:     "eligible",
+			eligible: true, connected: true, funded: true, answersDeviate: true,
+			expectedToPoll: true, expectedToSubmit: true,
+		}, {
+			name:     "ineligible",
+			eligible: false, connected: true, funded: true, answersDeviate: true,
+			expectedToPoll: false, expectedToSubmit: false,
+		}, {
+			name:     "disconnected",
+			eligible: true, connected: false, funded: true, answersDeviate: true,
+			expectedToPoll: false, expectedToSubmit: false,
+		}, {
+			name:     "under funded",
+			eligible: true, connected: true, funded: false, answersDeviate: true,
+			expectedToPoll: false, expectedToSubmit: false,
+		}, {
+			name:     "answer undeviated",
+			eligible: true, connected: true, funded: true, answersDeviate: false,
+			expectedToPoll: true, expectedToSubmit: false,
+		}, {
+			name:     "previous job run completed",
+			eligible: true, connected: true, funded: true, answersDeviate: true,
+			hasPreviousRun: true, previousRunStatus: pipeline.RunStatusCompleted,
+			expectedToPoll: false, expectedToSubmit: false,
+		}, {
+			name:     "previous job run in progress",
+			eligible: true, connected: true, funded: true, answersDeviate: true,
+			hasPreviousRun: true, previousRunStatus: pipeline.RunStatusInProgress,
+			expectedToPoll: false, expectedToSubmit: false,
+			// There doesn't seem to be a way to check for cancelled
```

Comment:
There doesn't seem to be a way to determine if a pipeline run has been cancelled, so I've let this out for now

Reply:
Removed the test